robottelo/hosts.py

```diff
@@ -430,20 +519,252 @@ def set_infrastructure_type(self, infrastructure_type='physical'):
         script_content = '\n'.join(script_content)
         self.execute(f"echo -e '{script_content}' > {script_path}")
 
-    def patch_os_release_version(self, distro=DISTRO_RHEL7):
+    def patch_os_release_version(self, distro=constants.DISTRO_RHEL7):
         """Patch VM OS release version.
 
         This is needed by yum package manager to generate the right RH
         repositories urls.
         """
-        if distro == DISTRO_RHEL7:
-            rh_product_os_releasever = REPOS['rhel7']['releasever']
+        if distro == constants.DISTRO_RHEL7:
+            rh_product_os_releasever = constants.REPOS['rhel7']['releasever']
         else:
             raise ContentHostError('No distro package available to retrieve release version')
         return self.execute(f"echo '{rh_product_os_releasever}' > /etc/yum/vars/releasever")
 
+    # What the heck to call this function?
+    def host_populate(
+        self,
+        org_label,
+        rh_repos_id=None,
+        repos_label=None,
+        product_label=None,
+        lce=None,
+        activation_key=None,
+        patch_os_release_distro=None,
+        install_katello_agent=True,
+    ):
+        """
+        Setup a Content Host with basic components and tasks.
+
+        :param str org_label: The Organization label.
+        :param list rh_repos_id: a list of RH repositories ids to enable.
+        :param list repos_label: a list of custom repositories labels to enable.
+        :param str product_label: product label if repos_label is applicable.
+        :param str lce: Lifecycle environment label if applicable.
+        :param str activation_key: Activation key name if applicable.
+        :param str patch_os_release_distro: distro name, to patch the VM with os version.
+        :param bool install_katello_agent: whether to install katello agent.
+        """
+        rh_repos_id = rh_repos_id or []
+        repos_label = repos_label or []
+        self.install_katello_ca()
+        self.register_contenthost(org_label, activation_key=activation_key, lce=lce)
+        if not self.subscribed:
+            raise CLIFactoryError('Virtual machine failed subscription')
+        if patch_os_release_distro:
+            self.patch_os_release_version(distro=patch_os_release_distro)
+        # Enable RH repositories
+        for repo_id in rh_repos_id:
```

Comment:
Good eye, but there's a statement to deal with this at the very start of the function:

```
rh_repos_id = rh_repos_ids or []
```

If nothing was passed and the default `None` from the function signature is used, this statement will replace it with an empty list, to avoid the TypeError here.

Reply:
btw that statement is using `rh_repos_ids` because I had half updated things from your suggestion below to use plural keyword names, but that's beside the point when it comes to the potential for a TypeError there.

Reply:
Ah I didn't notice that, nicely done